File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/AbstractUserPermissionCommand.java
package org.dddml.wms.domain;

import java.util.*;
import java.util.Date;
import org.dddml.wms.domain.AbstractCommand;

public abstract class AbstractUserPermissionCommand extends AbstractCommand implements UserPermissionCommand
{
    private String permissionId;

    public String getPermissionId()
    {
        return this.permissionId;
    }

    public void setPermissionId(String permissionId)
    {
        this.permissionId = permissionId;
    }

    private String userId;

    public String getUserId()
    {
        return this.userId;
    }

    public void setUserId(String userId)
    {
        this.userId = userId;
    }


    public static abstract class AbstractCreateOrMergePatchUserPermission extends AbstractUserPermissionCommand implements CreateOrMergePatchUserPermission
    {
        private Boolean active;

        public Boolean getActive()
        {
            return this.active;
        }

        public void setActive(Boolean active)
        {
            this.active = active;
        }

    }

    public static abstract class AbstractCreateUserPermission extends AbstractCreateOrMergePatchUserPermission implements CreateUserPermission
    {
        @Override
        public String getCommandType() {
            return COMMAND_TYPE_CREATE;
        }

    }

    public static abstract class AbstractMergePatchUserPermission extends AbstractCreateOrMergePatchUserPermission implements MergePatchUserPermission
    {
        @Override
        public String getCommandType() {
            return COMMAND_TYPE_MERGE_PATCH;
        }

        private Boolean isPropertyActiveRemoved;

        public Boolean getIsPropertyActiveRemoved()
        {
            return this.isPropertyActiveRemoved;
        }

        public void setIsPropertyActiveRemoved(Boolean removed)
        {
            this.isPropertyActiveRemoved = removed;
        }

    }

    public static class SimpleCreateUserPermission extends AbstractCreateUserPermission
    {
    }

    public static class SimpleMergePatchUserPermission extends AbstractMergePatchUserPermission
    {
    }

	public static class SimpleRemoveUserPermission extends AbstractUserPermissionCommand implements RemoveUserPermission
	{
        @Override
        public String getCommandType() {
            return COMMAND_TYPE_REMOVE;
        }
	}


}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/AbstractUserPermissionCommand.php
<?php

namespace Dddml\Wms\Domain;

use Dddml\Command\CommandInterface;
use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\Wms\Domain\CommandTrait;

abstract class AbstractUserPermissionCommand implements CommandInterface
{

    use CommandTrait;

    /**
     * @Type("string")
     */
    private $permissionId;

    /**
     * @return string
     */
    public function getPermissionId()
    {
        return $this->permissionId;
    }

    /**
     * @param string $permissionId
     */
    public function setPermissionId($permissionId)
    {
        $this->permissionId = $permissionId;
    }


}

